README.md

```diff
@@ -147,61 +147,18 @@ https://github.com/angular/angular-cli/issues/6349).
 
 ### Updating the API using swagger-codegen
 
-We use [swagger-codegen](https://github.com/swagger-api/swagger-codegen) to automatically implement the API, as defined in `api/jobs.yaml`, for all
-servers and the UI. Whenever the API is updated, follow these steps to update the server implementations. 
-
-**Note:** after updating the API files you will need to test and update the server implementations to resolve any broken dependencies on old 
-API definitions or implement additional functionality to match the new specs. 
-
-From the base of the checked-out job-manager repository, run:
-```sh
-scripts/rebuild_swagger.sh
-```
-
-This will:
- - Find and download `swagger-codegen-cli.jar`
- - Remove the necessary build artifacts 
- - Use `swagger-codegen-cli.jar` to regenerate them. 
-
-If you prefer to do this manually you can fallow the instructions below.
-
-#### Manual Process
+We use [swagger-codegen](https://github.com/swagger-api/swagger-codegen) to automatically generate the API classes
+defined in `api/jobs.yaml` for the servers and for the UI. 
+Whenever the API is updated, re-run `docker-compose up rebuild-swagger` to trigger a rebuild.
+ 
+**Note:** the `rebuild-swagger` job does nothing if the file `api/jobs.yaml` has not changed since the
+last time it was run.
 
-If you prefer to perform the steps manually, you can:
+**Note:** the `rebuild-swagger` job will run by default during `docker-compose up` to generate the swagger for the other 
+services if necessary. The other services will not start until their swagger classes exist.
 
-1. If you do not already have the jar, you can download it here:
- 
-    ```
-      # Using wget:
-      wget http://central.maven.org/maven2/io/swagger/swagger-codegen-cli/2.2.3/swagger-codegen-cli-2.2.3.jar -O swagger-codegen-cli.jar
-      
-      # Or, using curl:
-      curl http://central.maven.org/maven2/io/swagger/swagger-codegen-cli/2.2.3/swagger-codegen-cli-2.2.3.jar > swagger-codegen-cli.jar
-    ```
-    
-2. Clear out existing generated models:
-    ```
-    rm ui/src/app/shared/model/*
-    rm servers/dsub/jobs/models/*
-    rm servers/cromwell/jobs/models/*
-    ```
-3. Regenerate both the python and angular definitions.
-    ```
-    java -jar swagger-codegen-cli.jar generate \
-      -i api/jobs.yaml \
-      -l typescript-angular2 \
-      -o ui/src/app/shared
-      java -jar swagger-codegen-cli.jar generate \
-      -i api/jobs.yaml \
-      -l python-flask \
-      -o servers/dsub \
-      -DsupportPython2=true,packageName=jobs
-      java -jar swagger-codegen-cli.jar generate \
-      -i api/jobs.yaml \
-      -l python-flask \
-      -o servers/cromwell \
-      -DsupportPython2=true,packageName=jobs
-      ```
+**Note:** after regenerating the model files, you'll need to test and update the server implementations to 
+resolve any broken dependencies on old API definitions or implement additional functionality to match the new specs. 
```

Comment:
nit: I'd change these **Note:** 's to one "Notes" header and a list (see https://github.com/DataBiosphere/job-manager/blob/master/README.md#notes)